pyqtorch/modules/hamevo.py

```diff
@@ -165,47 +166,23 @@ def apply(self, state: torch.Tensor) -> torch.Tensor:
 
         return _apply_batch_gate(state, evol_operator, self.qubits, self.n_qubits, batch_size_h)
 
+class HamEvoType(Enum):
+    RK4 = HamEvo
+    EIG = HamEvoEig
+    EXP = HamEvoExp
 
 class HamiltonianEvolution(Module):
-    def __init__(self, qubits: Any, n_qubits: int, n_steps: int = 100):
+    def __init__(self, qubits: Any, n_qubits: int, n_steps: int = 100, type: HamEvoType = HamEvoType.RK4):
```

Comment:
The type of the `type` argument should be `HamEvoType | str` to be consistent. Furthermore, it is better to avoid naming the variable `type` since it clashes with the Python standard library function. Maybe `hamevo_type`?